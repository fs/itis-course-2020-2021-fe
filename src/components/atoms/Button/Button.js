import styled, { css } from 'styled-components';

const StyledButton = styled.button(
  ({ primary, error, outlined = true }) => css`
    color: ${primary ? 'white' : 'blue'};
    ${error && 'color: red'};
    background-color: ${primary ? 'blue' : 'white'};
    font-size: 16px;
    border: 1px solid blue;
    ${error && 'border: 1px solid red'};
    padding: 4px 8px;
    cursor: pointer;

    :disabled {
      border: 1px solid gray;
      background-color: grey;
      color: white;
      cursor: auto;
    }

    :focus {
      outline: #61dafb 1px solid;
    }

    :active {
      outline: #61dafb 1px solid;
    }

    ${!outlined &&
    css`
      border: 0;

      :focus {
        outline: 0;
      }

      :active {
        outline: 0;
      }
    `};
  `,
);

const Button = (props) => {
  const { children, ...otherProps } = props;
  return <StyledButton {...otherProps}>{children}</StyledButton>;
};

export default Button;
