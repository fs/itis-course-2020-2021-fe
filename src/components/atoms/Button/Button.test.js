import { render, screen } from '@testing-library/react';
import Button from './Button';

test('renders button', () => {
  render(<Button>Click</Button>);
  const linkElement = screen.getByText('Click');
  expect(linkElement).toBeInTheDocument();
});
