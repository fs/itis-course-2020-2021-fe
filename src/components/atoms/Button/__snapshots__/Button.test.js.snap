// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should not call function onClick on disabled button 1`] = `
<button
  class="Button__StyledButton-sc-1wq17d6-0 jTfwWo"
  disabled=""
>
  Click
</button>
`;
